float64[7] qc2
float64[7] qc1
float64[7] tau_d